lms/djangoapps/course_structure_api/v0/tests.py

```diff
@@ -375,3 +401,162 @@ def test_get(self):
+        """
+        super(CourseBlocksOrNavigationMixin, self).setUp()
+        CourseEnrollmentFactory(user=self.user, course_id=self.course.id)
+
+    def create_user(self):
+        """
+        Override the base `create_user` method to test with non-staff users for these views.
+        """
+        self.user = UserFactory.create()
+
+    @property
+    def view(self):
+        """
+        Returns the name of the view for testing to use in the django `reverse` call.
+        """
+        return 'course_structure_api:v0:' + self.block_navigation_view_type
+
+    def test_get(self):
+        with check_mongo_calls(3):
+            response = super(CourseBlocksOrNavigationMixin, self).test_get()
+
+        # verify root element
+        self.assertIn('root', response.data)
+        root_string = unicode(self.course.location)
+        self.assertEquals(response.data['root'], root_string)
+
+        # verify ~blocks element
+        self.assertTrue(self.block_navigation_view_type in response.data)
+        blocks = response.data[self.block_navigation_view_type]
+
+        # verify number of blocks
+        self.assertEquals(len(blocks), 4)
+
+        # verify fields in blocks
+        for field, block in product(self.block_fields, blocks.values()):
+            self.assertIn(field, block)
+
+        # verify container fields in container blocks
+        for field in self.container_fields:
+            self.assertIn(field, blocks[root_string])
+
+    def test_parse_error(self):
+        """
+        Verifies the view returns a 400 when a query parameter is incorrectly formatted.
+        """
+        response = self.http_get_for_course(data={'block_json': 'incorrect'})
+        self.assertEqual(response.status_code, 400)
+
+    def test_no_access_to_block(self):
+        """
+        Verifies the view returns only the top-level course block, excluding the sequential block
+        and its descendants when the user does not have access to the sequential.
+        """
+        self.sequential.visible_to_staff_only = True
+        modulestore().update_item(self.sequential, self.user.id)
+
+        response = super(CourseBlocksOrNavigationMixin, self).test_get()
+        self.assertEquals(len(response.data[self.block_navigation_view_type]), 1)
+
+
+class CourseBlocksMixin(object):
+    """
+    A Mixin class for testing all views related to Course blocks.
+    """
+    view_supports_debug_mode = False
+    block_fields = ['id', 'type', 'display_name', 'web_url', 'block_url', 'graded', 'format']
+
+    def test_block_json(self):
+        """
+        Verifies the view's response when the block_json data is requested.
+        """
+        response = self.http_get_for_course(
+            data={'block_json': '{"video":{"profiles":["mobile_low"]}}'}
+        )
+        self.assertEquals(response.status_code, 200)
+        video_block = response.data[self.block_navigation_view_type][unicode(self.video.location)]
```

Comment:
The convention of naming a class as a Mixin implies that it is abstract and intended to be inherited from.  However, I can see that this could be made more explicit.  I added additional documentation preceding the Test classes to make it clear which attributes should be defined in the concrete classes.  I also explicitly marked the Mixins as "abstract" (with `ABC`).  Thanks.